src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1245,63 +1236,67 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
                 // Start the write.  This serializes access to write to the connection, and ensures that HEADERS
                 // and CONTINUATION frames stay together, as they must do. We use the lock as well to ensure new
                 // streams are created and started in order.
-                Memory<byte> writeBuffer = await StartWriteAsync(totalSize, cancellationToken).ConfigureAwait(false);
-                try
+                await InvokeLockedAsync(totalSize, (thisRef: this, http2Stream, current, remaining, totalSize, flags, mustFlush), (s, writeBuffer) =>
```

Comment:
That's a lot of state. Is the JIT able to enregister all of that in the callback? If not, is it worth it to copy them to individual variables?

Reply:
Last I checked, I believe none would be enregistered, but it's possible that's changed with some of @CarolEidt's struct-related changes?

Reply:
The changes in flight now, with #36862 and upcoming changes from @sandreenko will only impact the JIT's ability to keep value types in registers when they're passed or returned in registers.

Reply:
Thanks, @caroleidt.